### DISCUSSION TITLE
Welcome to nestjs-hybrid-auth Discussions!

### DISCUSSION THREAD

**Discussion by mjangir** [mjangir/nestjs-hybrid-auth#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mb38** [comment]
Hello, I don't understand how to use the library.
On the frontend, you need to set a link to the endpoint 
for example:
`<a href="http://localhost:3000/auth/google">Login Google</a>`
The backend via @UseGoogleAuth() redirects the user to the Google authorization page, the user confirms access and is redirected to http://localhost:3000/auth/google-login/callback

```
export class AppController {
  @UseGoogleAuth()
  @Get('auth/google')
  loginWithGoogle() {
    return 'Login with Google';
  }

  @UseGoogleAuth()
  @Get('auth/google-login/callback')
  googleCallback(@Req() req) {
    const result: GoogleAuthResult = req.hybridAuthResult;

    req.res.setHeader('Set-Cookie', [result.accessToken, result.refreshToken]);

    return {
      accessToken: result.accessToken,
      refreshToken: result.refreshToken,
      profile: result.profile,
    };
  }
}
```

I don't understand how can I notify the frontend that the user has confirmed access in the Google account?
Or what should I do as a backend developer to provide this information?

**Comment by rrawann** [comment]
Hello
When signUp with twitter and I have my consumerKey and consumerSecret , but i got Internal Server 500 "message":"Could not authenticate you." how can fix it?
